sklearn/tests/test_common.py

```diff
@@ -356,6 +360,7 @@ def test_set_output_transform(estimator):
             f"Skipping check_set_output_transform for {name}: Does not support"
             " set_output API"
         )
+    estimator = next(_yield_instances_for_check(check_set_output_transform, estimator))
```

Comment:
seems to be a nice catch, but I think we should then go through all instances for this test, instead of only the first one.